core/src/main/java/org/mskcc/cbio/portal/authentication/aaf/AafJwtHandler.java

```diff
@@ -0,0 +1,104 @@
+package org.mskcc.cbio.portal.authentication.aaf;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.MACVerifier;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import java.text.ParseException;
+import java.util.Date;
+
+/**
+ * Responsible for handling and validating assertions from AAF Rapid Connect
+ * 
+ * Created by Shaun Muscat
+ */
+class AafJwtHandler {
+    
+    private static final Log log = LogFactory.getLog(PortalUserDetailsService.class);
+    
+    private final SignedJWT signedJWT;
+    private final String issuer;
+    private final String primaryURL;
+    private final String sharedSecret;
+
+    /**
+     * Constructor
+     * <p/>
+     * Takes the signed JWT (JWS) assertion received from AAF Rapid Connect in addition to some details about the 
+     * registered service, and performs validation of the assertion.
+     * 
+     * @param assertion signed JWT (JWS) assertion received from AAF Rapid Connect
+     * @param sharedSecret shared secret between AAF Rapid Connect and the application for JWT signing and verification
+     * @param primaryURL primary URL of application, provided as part of service registration
+     * @param productionFederation true if the service was registered to the production federation, false otherwise
+     * @throws Exception thrown if signed JWT (JWS) validation failed or could not be parsed
+     */
+    AafJwtHandler(String assertion, String sharedSecret, String primaryURL, Boolean productionFederation) 
+        throws Exception {
+        this.sharedSecret = sharedSecret;
+        this.primaryURL = primaryURL;
+        if (productionFederation) {
+            this.issuer = "https://rapid.aaf.edu.au";
```

Comment:
In general it's bad to have test environment checks in code, having it as a parameter makes it more general and robust. You could have multiple any number of environments if you wanted (unlikely in this). Besides, it would reduce 5 lines of code to just one: `this.issuer = issuer`